### PR TITLE
Frontend Feature: Multiselect ui-component and merge refs hook

### PR DESCRIPTION
**Pull request by 596050**
New MultiSelect Component

I've been trying out some new langflow components I'd like to add to the codebase but they need something like a multi select ui-component which appears to be unavailable. Currently no tests because there appear to be end-to-end tests only which requires the ui-component to be in a langflow component, currently no langflow components have this ui-component in it. Would be great to discuss.

Usage:

```jsx
<MultiSelect
  disabled={disabled}
  options={[
    { value: "option_1", label: "This is an extra extra extra long label to shsow responsiveness of the component" },
    { value: "option_2", label: "Option A" },
    { value: "option_3", label: "Option B" },
    { value: "option_4", label: "Option C" },
    { value: "option_5", label: "Option D" },
  ]}
  value={data?.node?.template?.[name]?.value || []}
  id={"multiselect-" + name}
  onValueChange={handleOnNewValue}
/>
```

User experience:
https://github.com/langflow-ai/langflow/assets/19141517/cffd9d64-414a-4ac8-8008-5a8ad2797546

Screenshot 2024-06-27 at 16 24 44

Screenshot 2024-06-27 at 16 24 53

Also adds a useMergeRefs custom hook

### CODE REVIEW

**Comment by ogabrielluiz** [comment]
Wow! That's really awesome and really needed. 

Thank you so much @596050

**Comment by ogabrielluiz** [comment]
I can add support for it on the Component in no time.

**Comment by lucaseduoli** [comment]
Hello!

I loved this component. Can you implement it on the ParameterComponent and the TableNodeCellRenderer please?
The conditions to show the component are going to be:

- Type is "str"
- template[name].options exist
- template[name].list is true

 We're going to implement the backend components to use that as soon as possible.

**Review by lucaseduoli — CHANGES_REQUESTED** [review]
Hello!

I loved this component. Can you implement it on the ParameterComponent and the TableNodeCellRenderer please?
The conditions to show the component are going to be:

Type is "str"
template[name].options exist
template[name].list is true
We're going to implement the backend components to use that as soon as possible.

File: src/backend/base/langflow/inputs/inputs.py [hunk 1]
```diff
@@ -289,6 +290,23 @@ class DropdownInput(BaseInputMixin, DropDownMixin, MetadataTraceMixin):
     options: list[str] = Field(default_factory=list)
 
 
+class MultiselectInput(BaseInputMixin, MultiselectMixin, MetadataTraceMixin):
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/inputs/inputs.py:293]:
```suggestion
class MultiselectInput(BaseInputMixin, ListableMixin, DropDownMixin, MetadataTraceMixin):
    field_type: Optional[SerializableFieldTypes] = FieldTypes.TEXT
```

I think this should be enough to create this type.

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Hey @596050 

Just this fix and we can merge it.

**Pull request merged**